querybook/webapp/ui/StyledText/StyledText.tsx

```diff
@@ -0,0 +1,220 @@
+import clsx from 'clsx';
+import * as React from 'react';
+import styled from 'styled-components';
+
+interface IStyledTextProps {
+    weight?: 'normal' | 'light' | 'bold' | 'extra';
+    size?:
+        | 'xxxsmall'
+        | 'xxsmall'
+        | 'xsmall'
+        | 'small'
+        | 'text'
+        | 'text-0'
+        | 'med'
+        | 'large'
+        | 'xlarge'
+        | 'xxlarge'
+        | 'xxxlarge';
+    color?:
+        | 'lightest-0'
+        | 'lightest'
+        | 'light'
+        | 'text'
+        | 'dark'
+        | 'invert'
+        | 'accent'
+        | 'accent-dark';
+    cursor?: 'default' | 'pointer' | 'not-allowed';
+
+    accent?: boolean;
+    noUserSelect?: boolean;
+    hover?: boolean;
+
+    untitled?: boolean;
+    center?: boolean;
+}
+
+interface IProps extends IStyledTextProps {
+    className?: string;
+
+    tooltip?: string;
+    tooltipPos?: 'up' | 'left' | 'right' | 'down';
+}
+
+const StyledTextSpan = styled((props: IStyledTextProps) => <div {...props} />)`
+    ${(props) =>
+        props.weight
+            ? ` font-weight: ${
+                  props.weight === 'bold'
+                      ? 'var(--bold-font)'
+                      : `var(--${props.weight}-bold-font)`
+              }`
+            : ''};
+    ${(props) =>
+        props.size
+            ? ` font-size: ${
+                  props.size === 'text'
+                      ? 'var(--text-size)'
+                      : `var(--${props.size}-text-size)`
+              }`
+            : ''};
+    ${(props) =>
+        props.color
+            ? ` color: ${
+                  props.color === 'text'
+                      ? 'var(--text)'
+                      : props.color.startsWith('accent')
+                      ? `var(--color-${props.color})`
+                      : `var(--text-${props.color})`
+              }`
+            : ''};
+    ${(props) =>
+        props.accent
```

Comment:
accentFont

Reply:
this is just a minor suggestion, I found it confusing to differentiate between color accent and font accent 😂